### PR TITLE
DROP BEFORE MERGE: test petsc update

### PR DESCRIPTION
**Pull request by wence-**
Just testing that updating petsc doesn't break anything.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -40,7 +40,7 @@ pipeline {
               sh 'mkdir tmp'
               dir('tmp') {
                 timestamps {
-                  sh '../scripts/firedrake-install $COMPLEX --tinyasm --disable-ssh --minimal-petsc --slepc --documentation-dependencies --install thetis --install gusto --install icepack --install irksome --no-package-manager || (cat firedrake-install.log && /bin/false)'
+                  sh '../scripts/firedrake-install $COMPLEX --package-branch petsc wence/updates --tinyasm --disable-ssh --minimal-petsc --slepc --documentation-dependencies --install thetis --install gusto --install icepack --install irksome --no-package-manager || (cat firedrake-install.log && /bin/false)'
```
Comment (kaushikcfd) [inline Jenkinsfile:43]: Now that petsc4py is a part of petsc's tree, `firedrakeproject/petsc4py@firedrake` wouldn't be tracking the latest changes at https://gitlab.com/petsc/petsc/-/tree/master/src/binding/petsc4py.

Comment (wence-) [reply Jenkinsfile:43]: Yeah. We need to change the way we install petsc4py to get it from the petsc repo

Comment (kaushikcfd) [reply Jenkinsfile:43]: A possible fix is:
```diff
diff --git a/scripts/firedrake-install b/scripts/firedrake-install
index f3a9e8b2..f4163f0c 100755
--- a/scripts/firedrake-install
+++ b/scripts/firedrake-install
@@ -635,6 +635,12 @@ def git_clone(url):
     if name == "petsc" and args.honour_petsc_dir:
         log.info("Using existing petsc installation\n")
         return name
+    if name == "petsc4py":
+        # petsc4py not a repo: copy from petsc_dir
+        petsc_dir, _ = get_petsc_dir()
+        petsc4py_dir = os.path.join(petsc_dir, "src", "binding", "petsc4py")
+        check_call(["cp", "-r", petsc4py_dir, 'petsc4py/'])
+        return name
     log.info("Cloning %s\n" % name)
     branch = branches.get(name.lower(), branch)
     try:
```
After a deprecation period, we might get rid of repo. `firedrakeproject/petsc4py` completely.

Comment (wence-) [reply Jenkinsfile:43]: Why copy instead of just installing directly from the bindings subdir?

Comment (kaushikcfd) [reply Jenkinsfile:43]: Yep, that's better.

```diff
diff --git a/scripts/firedrake-install b/scripts/firedrake-install
index f3a9e8b2..541cc87d 100755
--- a/scripts/firedrake-install
+++ b/scripts/firedrake-install
@@ -635,6 +635,9 @@ def git_clone(url):
     if name == "petsc" and args.honour_petsc_dir:
         log.info("Using existing petsc installation\n")
         return name
+    if name == "petsc4py":
+        # petsc4py not a repo: present at petsc_dir/src/binding
+        return name
     log.info("Cloning %s\n" % name)
     branch = branches.get(name.lower(), branch)
     try:
@@ -768,7 +771,11 @@ def install(package):
     elif package == "slepc/":
         build_and_install_slepc()
     # The following outrageous hack works around the fact that petsc and co. cannot be installed in developer mode.
-    elif package not in ["petsc4py/", "slepc4py/"]:
+    elif package == "petsc4py":
+        petsc_dir, _ = get_petsc_dir()
+        petsc4py_dir = os.path.join(petsc_dir, "src", "binding", "petsc4py")
+        run_pip_install(["-e", petsc4py_dir])
+    elif package not in ["slepc4py/"]:
         run_pip_install(["-e", package])
     else:
         run_pip_install(["--ignore-installed", package])
```

**Comment by wence-** [comment]
Update, it did break things...

**Pull request closed without merging**